### PR TITLE
ARROW-21 Test PyMongoArrow documentation build in CI

### PR DESCRIPTION
**Pull request by blink1073**
Also addresses ARROW-41

### CODE REVIEW

File: .github/workflows/test-python.yml [hunk 1]
```diff
@@ -25,10 +25,8 @@ jobs:
     name: CPython ${{ matrix.python-version }}-${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
-      - name: Setup Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.python-version }}
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
```
Comment (blink1073) [inline .github/workflows/test-python.yml:29]: This action takes care of installing python and setting up caching for pip and conda

File: bindings/python/setup.py [hunk 2]
```diff
@@ -38,43 +38,52 @@ def query_pkgconfig(cmd):
...
-
-        # Find the linkable library file, and explicity add it to the linker if on Windows.
-        lib_dirs = glob.glob(os.path.join(install_dir, "lib*"))
-        if len(lib_dirs) != 1:
-            warnings.warn(f"Unable to locate libbson in {install_dir}")
+    dynamic_lib_dir = 'bin' if IS_WIN else 'lib*'
+    static_lib_dir = 'lib*'
+
+    # If no install dir is given, try looking for an installed python
+    # package (e.g. libbson from conda-forge).
```
Comment (blink1073) [inline bindings/python/setup.py:45]: This will also be useful if we create a libbson package and publish to PyPI.

**Comment by blink1073** [comment]
Moving back to draft so I can create the appropriate RTD config file.

**Review by ShaneHarvey — COMMENTED** [review]
There are surprisingly a lot of changes to setup.py here. Can you explain?

**Comment by blink1073** [comment]
> There are surprisingly a lot of changes to setup.py here. Can you explain?

I was refactoring to support installing from the `libbson` `conda` package without requiring an environment variable (since we can't set it on RTD).  I also wanted to preserve the ability to use the system `libbson` (I have the `homebrew` version installed locally).  I noticed that the `IS_WIN` vs Posix handling branch were getting muddled, so I separated it out into different functions, which allowed me to clean it up a lot.

**Comment by ShaneHarvey** [comment]
Interesting. Is supporting libbson conda needed? It seems like we could just run `./build-libbson.sh` like usual before building the docs.

**Comment by blink1073** [comment]
> It seems like we could just run ./build-libbson.sh like usual before building the docs.

ReadTheDocs does not yet support running scripts prior to `pip install .`.  However, I found the solution: if you install `pkg-config` in the `conda` environment, then that version of `pkg-config` looks in the conda environment for its libraries, and it works with no changes to `setup.py`!

File: .readthedocs.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+version: 2
+sphinx:
+  configuration: ./bindings/python/docs/source/conf.py
+conda:
+  environment: ./bindings/python/docs/environment.yml
+python:
+  install:
+    # install the package itself
+    - method: pip
+      path: ./bindings/python
```
Comment (ShaneHarvey) [inline .readthedocs.yaml:10]: Does this build properly properly on readthedocs now that the workarounds are removed? If so do we need jupyterlab at all?

Comment (blink1073) [reply .readthedocs.yaml:10]: We're only using JupyterLab Maintainer-Tools to simplify the Actions logic, it saves us from having to keep all those things up to date and in sync.

Comment (blink1073) [reply .readthedocs.yaml:10]: The PR build works!

Comment (blink1073) [reply .readthedocs.yaml:10]: Removed usage of the jupyterlab action

File: bindings/python/docs/environment.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: mongo_arrow_documentation
+channels:
+  - conda-forge
+dependencies:
+- python=3.9
+- sphinx
+- libbson
```
Comment (ShaneHarvey) [inline bindings/python/docs/environment.yml:7]: no longer needed?

**Review by ShaneHarvey — APPROVED** [review]
LGTM with one minor comment

File: bindings/python/docs/environment.yml (hunk 4 above)
Comment (blink1073) [reply bindings/python/docs/environment.yml:7]: We are still using it, as the "system" `libbson`

Comment (ShaneHarvey) [reply bindings/python/docs/environment.yml:7]: Thanks!

**Pull request merged**